app/controllers/api/v2/moves_actions.rb

```diff
@@ -13,18 +13,16 @@ def show_and_render
 
     def create_and_render
       move = Move.new(move_attributes)
-      move.determine_supplier
-      authorize!(:create, move)
-      move.save!
-
+      move.determine_supplier(doorkeeper_application_owner)
+      move.save! # NB: the move should be saved and notified even if the current user is no longer authorised to view it
       Notifier.prepare_notifications(topic: move, action_name: 'create')
-
+      authorize!(:create, move)
```

Comment:
Seems very confusing for a consumer of this api to get a failure message and actually succeed in creating a move 😕 .